File: Destro/Destro/Interfaces/IMachines.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace BioDieselProject.Interfaces
{
    public abstract class IMachines
    {
        public double Capacity{ get; set; }

        public double Flow{ get; set; }

        public double Volume { get; set; }

        public double Waste { get; set; }

        public virtual object setCapacity(double quantity)
        {
            Capacity += quantity;

            return new { Capacity };
        }

        public virtual double trasfer()
        {
            double transfer = 0;
            if (Capacity > 0)
            {
                if (Capacity <= Flow)
                {
                    transfer = Capacity;
                    Capacity -= transfer;
                }
                else
                {
                    transfer = Flow;
                    Capacity -= transfer;
                }
            }
            return transfer;
        }
        
    }
}

File: Interface/IMachines.py
import socket
import time


class IMachines:
    def __init__(self):
        self.Capacity = 0
        self.Flow = 0
        self.Volume = 0
        self.Waste = 0

    def setCapacity(self, quantity):
        self.Capacity += quantity

    def calculateTransfer(self, restante=""):
        transfer = 0

        if (self.Capacity > 0):
            sizeSubstance = self.Capacity if self.Capacity <= self.Flow else self.Flow
            if restante != '':
                quantidadeASerTransferida = restante if restante <= sizeSubstance else sizeSubstance
                transfer = quantidadeASerTransferida
                self.Capacity -= transfer
            else:
                transfer = sizeSubstance
                self.Capacity -= transfer
        return transfer
